### PR TITLE
update protos and remove sensorinfo, deviceinfo and activity status setters in favor of getBuilder

### PR DESCRIPTION
**Pull request by FabianTerhorst**

### CODE REVIEW

File: library/src/main/java/com/pokegoapi/api/device/LocationFixes.java [hunk 1]
```diff
@@ -107,7 +107,11 @@ public static LocationFixes getDefault(PokemonGo api, RequestEnvelopeOuterClass.
 					.setAltitude(altitude)
 					.setVerticalAccuracy(verticalAccuracy)
 					.setProviderStatus(3)
-					.setLocationType(1);
+					.setLocationType(1)
+					.setAltitude(api.getAltitude())
+					.setSpeed(/*Todo : speed iOS only (-1 for no reading available, speed in m/s)*/0)
+					.setCourse(/*Todo : iOS only (range seems to be -1 for not reading, and 0 to 360 for reading*/0)
+					.setFloor(/*Todo : The floor of the building this person is on*/0);
```
Comment (FabianTerhorst) [inline library/src/main/java/com/pokegoapi/api/device/LocationFixes.java:114]: @iGio90 you want to fill the 3 new values as well?

Comment (iGio90) [reply library/src/main/java/com/pokegoapi/api/device/LocationFixes.java:114]: i was already looking at it

Comment (FabianTerhorst) [reply library/src/main/java/com/pokegoapi/api/device/LocationFixes.java:114]: you found something?

Comment (FabianTerhorst) [reply library/src/main/java/com/pokegoapi/api/device/LocationFixes.java:114]: setCourse is just a random between 0 and 360

**Comment by Grover-c13** [comment]
course & speed can be -1 as well, still valid.

**Comment by FabianTerhorst** [comment]
ok

**Comment by FabianTerhorst** [comment]
and floor? ^^

**Comment by iGio90** [comment]
yeah, but is still better to findout a valid logic to random a good value in a good range. I'll take it please! :)

**Comment by FabianTerhorst** [comment]
i can just create a random for 0-360

**Comment by iGio90** [comment]
I would like to check before if it's really a random between 0-360 or there is a logic that could say if (accel_x > X || accel_y < Y) (or similar) course = Z

**Comment by FabianTerhorst** [comment]
normally the course shouldn´t have any subjections from other values

**Comment by FabianTerhorst** [comment]
for the speed we just have to calculate the distance between the last lat and lng and next and adding them to a value. This value is for the distance the person travelled since app start and calc that down to sec.

File: library/src/main/java/com/pokegoapi/api/PokemonGo.java [hunk 2]
```diff
@@ -273,11 +294,14 @@ public AuthInfo getAuthInfo()
 	 *
 	 * @param latitude  the latitude
 	 * @param longitude the longitude
-	 * @param altitude the altitude
+	 * @param altitude  the altitude
 	 * @param accuracy  the accuracy
 	 */
 	public void setLocation(double latitude, double longitude, float altitude, double accuracy) {
 		if (latitude != this.latitude || longitude != this.longitude) {
+			if (this.latitude != Double.NaN && this.longitude != Double.NaN) {
```
Comment (AbandonedCart) [inline library/src/main/java/com/pokegoapi/api/PokemonGo.java]: The default for each is NaN, so unless someone explicitly sets one to "Double.NaN", any time the previous check is met, this check will always be true.

**Comment by AbandonedCart** [comment]
This is inconsistent with device info and activity status, plus it moves toward preventing the user from setting their own values. While many will opt for defaults, removing the ability to easily set them as the end user is a huge regression.

**Comment by FabianTerhorst** [comment]
you can still set the own properties. Just sensorInfo.getBuilder()

**Comment by FabianTerhorst** [comment]
we should do the same for activitystatus and deviceinfo

**Comment by FabianTerhorst** [comment]
Now its not inconsistent anymore

**Comment by FabianTerhorst** [comment]
@iGio90 whats your opinion?

**Comment by FabianTerhorst** [comment]
should i make floor = altitude?

File: library/src/main/java/com/pokegoapi/api/PokemonGo.java [hunk 3]
```diff
@@ -311,7 +366,7 @@ public void setLongitude(double value) {
 	 * @throws IllegalStateException if location has not been set
 	 */
 	public Map getMap() {
-		if (this.latitude == Double.NaN || this.longitude == Double.NaN) {
```
Comment (s7092910) [inline library/src/main/java/com/pokegoapi/api/PokemonGo.java:314]: Should keep this. As there is still a single setter for longitude. Unless you plan to set the single setters for any location related value to private.

**Pull request closed without merging**